src/app/GitUI/UserControls/RevisionGrid/QuickSearchProvider.cs

```diff
@@ -56,37 +55,47 @@ public void OnKeyPress(KeyPressEventArgs e)
 
             curIndex = curIndex >= 0 ? curIndex : 0;
 
-            if (e.KeyChar == 8 && _quickSearchString.Length > 1)
+            if (e.KeyChar == (char)Keys.Back && _quickSearchString.Length > 1)
             {
                 // backspace
-                RestartQuickSearchTimer();
-
-                _quickSearchString = _quickSearchString[..^1];
-
-                FindNextMatch(curIndex, _quickSearchString, false);
-                _lastQuickSearchString = _quickSearchString;
-
-                e.Handled = true;
-                ShowQuickSearchString();
+                UpdateQuickSearchString(_quickSearchString[..^1]);
+            }
+            else if (e.KeyChar == 22 && Control.ModifierKeys == Keys.Control && Clipboard.ContainsText())
```

Comment:
❔ a little clearer
```suggestion
            else if (Control.ModifierKeys == Keys.Control && e.KeyChar == 'V' - '@' && Clipboard.ContainsText())
```

Reply:
👍 for changing the order of checks; as for the `'V' - '@` while google for "paste 22" I've found
this: https://stackoverflow.com/questions/12976062/is-there-a-risk-with-using-char22-in-code
and this: https://en.wikipedia.org/wiki/Control-V

so I think it's better to have a constant for it. I haven't met this char math while googling and it seems to raise more questions than it answers.